modules/dists/BlockDist.chpl

```diff
@@ -1953,6 +1959,17 @@ proc BlockDom.dsiLocalSubdomain(loc: locale) {
     return d;
   }
 }
+iter BlockDom.dsiLocalSubdomains(loc: locale) {
+  for locid in dist.chpl__locToLocIdxs(loc) {
+    var inds = chpl__computeBlock(locid, dist.targetLocDom, dist.boundingBox, dist.boundingBox.dims());
+    yield whole[(...inds)];
```

Comment:
I am surprised by this implementation. Shouldn't it be just `yield this.locDoms[locid].myBlock;`?

Reply:
Hmm, good question.  I took this approach due to the `else` clause in the `dsiLocalSubdomain()` routine just above, and to make it work with oversubscription; but in retrospect, I think you're right that I over-engineered it.  Specifically, I believe I was thinking that `myBlock` would only work in a non-oversubscribed environment, but with the loop over `locid`s, I think you're probably right.  I'll check that out to make sure that it works and doesn't introduce communication.

Reply:
Ooh, and I can update the tests to do the localAcesss calls within `local` blocks as a way to both verify this and lock it in going forward.

Reply:
> But then in my final review, I realized that the locale argument was never used / always defaulted to here, so removed that logic. So again, I think you're right.

Ah wait, it is necessary (if regrettable from a complexity standpoint) to do here since this routine takes a locale argument, so may be answering local subdomain queries about a remote locale's ownership, and we don't want to communicate with that remote locale to answer the question.

We could simplify / streamline the (common) `locale == here` case by either putting a conditional into this loop to check for it, or by creating two overloads of the routine, one which takes no argument (defaulting to `here`) and the other which doesn't—then updating ChapelArray.chpl's code to similarly split the implementation into two versions, each of which calls the corresponding one here.  I don't feel strongly inclined to do either of those things in this PR because I worry about keeping the two implementations in sync and would need to add more testing to distinguish between the remote and local cases.  It also feels a little like a change to the public interface (though it's one that I think would be acceptable since it wouldn't break existing code); or we could keep the same public interface and put a conditional into the outer call, but that doesn't seem great either.

That said, the `StencilDom.doiLocalStoredSubdomains` case should not be as expensive as I was fearing in the comment you commented on now that we don't accept a locale argument in that iterator, so that comment does need to be updated (and I can add a locality check to that test to lock that in as well).

Reply:
👍 I see. I agree that the solution you outlined would be good, but shouldn't be a part of this PR.